main.tf

```diff
@@ -0,0 +1,166 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.16"
+    }
+  }
+
+  required_version = ">= 1.5.7"
+
+  backend "s3" {
+    bucket               = "ccao-terraform-state-us-east-1"
+    key                  = "workspaces/default/terraform.tfstate"
+    region               = "us-east-1"
+    workspace_key_prefix = "workspaces"
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+locals {
+  s3_prefix                = "scripts/sales-val"
+  s3_bucket_data_warehouse = terraform.workspace == "prod" ? "ccao-data-warehouse-us-east-1" : aws_s3_bucket.data_warehouse[0].id
+  s3_bucket_glue_assets    = terraform.workspace == "prod" ? "ccao-glue-assets-us-east-1" : aws_s3_bucket.glue_assets[0].id
+  glue_job_name            = "sales_val_flagging${terraform.workspace == "prod" ? "" : "_${terraform.workspace}"}"
+  glue_crawler_name        = "ccao-data-warehouse-sale-crawler${terraform.workspace == "prod" ? "" : "-${terraform.workspace}"}"
```

Comment:
Jobs and crawlers will have name suffixes on staging environments that should make them easier to distinguish from their prod equivalents. If you have alternate suggestions for namespacing schemes, let me know!